R/functions-OnDiskMSnExp.R

```diff
@@ -232,11 +232,8 @@ precursorValue_OnDiskMSnExp <- function(object, column) {
     if (missing(fData) | missing(filenames))
         stop("Both 'fData' and 'filenames' are required!")
     filename <- filenames[fData[1, "fileIdx"]]
-    ## Open the file.
-    if (isCdfFile(filename))
-        fileh <- mzR::openMSfile(filename, backend = "netCDF")
-    else
-        fileh <- mzR::openMSfile(filename)
+    ## issue #214: define backend based on file format.
+    fileh <- mzR::openMSfile(filename, backend = mzRBackend(filename))
```

Comment:
Because you always use the same arguments for `openMSfile` (and the user is not allowed to change these and it is too easy to forget the `backend` argument), I would suggest to create a new function for this:
```r
.openMSfile <- function(filename) {
  mzR::openMSfile(filename, backend=mzRBackend(filename))
}
```